### ISSUE TITLE
[BUG] OneDrive: Could not locate the database file. Has it been renamed or moved?

### ISSUE THREAD

**Issue by poldixd** [strongbox-password-safe/Strongbox#791]
**Preliminaries**
I have ensured that:

- [X] I am running the latest version of Strongbox on the App Store by searching for Strongbox and clicking into it to see the Update button (or not)
- [X] I have performed a full restart of my device no matter how annoying that is

**On iOS (please complete the following information):**
 - Device: iPhone 13 Pro
 - OS: iOS 17.5.1

**Strongbox Version**
 - Version: 1.60.6

**Describe the bug**
Since today's update to 1.60.6, I can no longer synchronise my vault with OneDrive. I get the following error:

```
20:01:41.195 [In Bearbeitung] - Begin Sync [Interactive=YES, outstandingUpdate=YES, forcePull=0, provider=OneDrive, localModDate=2024-07-16 17:04:20.110, onlyIfModifiedDifferentFrom=2024-07-14 20:18:42.870, lastCheckedSourceMod=2024-07-14 20:18:42.870]
20:01:41.366 [Fehler] - Error Domain=com.markmcguill.strongbox. Code=45 "Could not locate the database file. Has it been renamed or moved?" UserInfo={NSLocalizedDescription=Could not locate the database file. Has it been renamed or moved?}
```

I can access the vault via the OneDrive web interface. I have tried it several times throughout the day in different wifis.

Can Strongbox with the latest update have a problem with umlauts (for example Ü) or special characters in the file names?

If I create a database without umlauts/special characters, everything works as usual.

The path to my vault is `700 - Schlüsselbund/Schlüsselbund.kdbx`.

**Comment by strongbox-mark** [comment]
Thanks for the bug report. We did make some big changes to OneDrive in this release to remove an old dependency and add support for shared/Sharepoint drives. So there's definitely a possibility we created a bug too!

However, I can't seem to reproduce this. I have created the following folder in the root of my OneDrive:

`700 - Schlüsselbund`

Using Strongbox I created a database in there called:

`Schlüsselbund.kdbx`

It seems to work and I can update etc. Is there anything else "funny" or "interesting" about the location or filename that I might be missing?

Is your OneDrive a business or personal one?

Is your Language / Region set to German?

Best wishes,
-Mark

**Comment by poldixd** [comment]
Hello Mark,

First of all, thank you for the great app and for the updates. I use it several times a day on my Mac and on my iPhone.

Have you changed anything with version 1.60.7? I've just installed the new version and it's working again.

**Comment by strongbox-mark** [comment]
Thanks @poldixd! 

Yes, we made a change or two in 1.60.7 that might have fixed it, but I'd much prefer to fully understand what it was that broke! :) 

Would you mind sending your debug info to support@strongboxsafe.com so I could take a look?

**Comment by unicorn855** [comment]
Good morning everyone!

I am encountering a potentially related synchronisation issue with OneDrive on the current version of strongbox as well. However, this error only occurs on my iPad (5. Generation) which can't upgrade further than iPadOS 16.7.8

Following is the error log when I try to open my database located in a sub-folder on my personal OneDrive:

09:47:25.038 [In Bearbeitung] - Begin Sync [Interactive=YES, outstandingUpdate=NO, forcePull=0, provider=OneDrive, localModDate=2024-07-13 17:52:01.337, onlyIfModifiedDifferentFrom=2024-07-13 17:52:01.337, lastCheckedSourceMod=2024-07-13 17:52:01.337]
09:47:25.060 [Fehler] - Strongbox.TwoDriveStorageProvider.TwoDriveStorageProviderError.couldNotGetDriveItemUrl


No errors whatsoever occur when I do the same thing on my iPhone13Pro running iOS 17.5.1 (21F90). 

Please let me know if I should make this a separate issue and/provide more debug info :-)

**Comment by strongbox-mark** [comment]
@unicorn855 - Can you update to 1.60.7 on the App Store now. Should fix your issue. In future, please include your App versions when reporting an issue. Let us know how it goes.

**Comment by unicorn855** [comment]
My apologies, when I wrote my first comment, I was already on app version 1.60.7 on all devices. I meant 'current' in the sense of current at the time of writing but I should have been more specific.

So no, at least for me, the issue is not fixed yet in version 1.60.7 (perhaps I should take this as a subtle hint of the universe telling me to get a current iPad :-D )

**Comment by poldixd** [comment]
> **[poldixd](/poldixd) ** c

The e-mail has been sent.

What else might be interesting: I firmly believe that the database name was changed from ‘Schlüsselbund’ to ‘Schlusselbund’ after the update. Could this be possible?

![Bildschirmfoto 2024-07-17 um 09 56 45](https://github.com/user-attachments/assets/38021fc2-c0a3-40f3-84e7-482b4d198ef3)

The screenshot was taken on a Mac M3 Pro Nov 2023 with Sonoma 14.5 and Strongbox 1.60.6. I noticed the same thing yesterday in the iPhone app.

**Comment by strongbox-mark** [comment]
@unicorn855 - Can you re-add your OneDrive database?

+ > Add Existing > OneDrive > ...

Does that work?

**Comment by unicorn855** [comment]
Already tried this, the actual adding works but then when I try to open the database and it attempts to sync again, the same error happens again.

**Comment by strongbox-mark** [comment]
Thanks @unicorn855 - Could you share your debug info to support@strongboxsafe.com

**Comment by strongbox-mark** [comment]
Just an update, we found an issue with older versions of iOS, anything less than iOS 17 and we've got a fix out in the App Store for that now as 1.60.8. Thanks to all for the help and info. Super strange one to do with URL query string parsing differences between iOS versions!
